cmp/form/FormModel.js

```diff
@@ -127,8 +132,9 @@ export class FormModel extends HoistModel {
     /**
      * Set the initial value of all fields and reset the form.
      *
-     * This is the main entry point for loading new data (or an empty new record) into the form for
-     * editing. If initial values are undefined for any field, existing initial values will be used.
+     * This is the main programmatic entry point for loading new data (or an empty new record)
+     * into the form for editing. If initial values are undefined for any field, the original
+     * initial values specified during model construction will be used.
```

Comment:
Great - this makes a lot of sense to me and seems like a smart way to get the behavior we expect!